src/main/java/com/mongodb/hibernate/jdbc/MongoResultSet.java

```diff
@@ -0,0 +1,282 @@
+        public int getColumnCount() {
+            return fieldNames.size();
+        }
+
+        @Override
+        public String getColumnLabel(final int column) {
+            return fieldNames.get(column - 1);
+        }
+    }
+
+    private final MongoCursor<BsonDocument> mongoCursor;
+
+    private final List<String> fieldNames;
+
+    private @Nullable BsonDocument currentDocument;
+
+    private @Nullable BsonValue lastReadColumnValue;
+
+    private boolean closed;
+
+    MongoResultSet(MongoCursor<BsonDocument> mongoCursor, List<String> fieldNames) {
+        this.mongoCursor = mongoCursor;
+        this.fieldNames = fieldNames;
+    }
+
+    @Override
+    public boolean next() throws SQLException {
+        checkClosed();
+        if (mongoCursor.hasNext()) {
+            currentDocument = mongoCursor.next().toBsonDocument();
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public void close() throws SQLException {
+        if (!closed) {
+            closed = true;
+            try {
+                mongoCursor.close();
+            } catch (RuntimeException e) {
+                throw new SQLException("Failed to close MongoCursor: " + e.getMessage(), e);
+            }
+        }
+    }
+
+    @Override
+    public boolean isClosed() {
+        return closed;
+    }
+
+    @Override
+    public boolean wasNull() throws SQLException {
+        checkClosed();
+        return assertNotNull(lastReadColumnValue).isNull();
+    }
+
+    @Override
+    public @Nullable String getString(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        return getValue(columnIndex, bsonValue -> bsonValue.asString().getValue());
+    }
+
+    @Override
+    public boolean getBoolean(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        return getValue(columnIndex, bsonValue -> bsonValue.asBoolean().getValue(), false);
+    }
+
+    @Override
+    public int getInt(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        return getValue(columnIndex, bsonValue -> bsonValue.asInt32().intValue(), 0);
+    }
+
+    @Override
+    public long getLong(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        return getValue(columnIndex, bsonValue -> bsonValue.asInt64().longValue(), 0L);
+    }
+
+    @Override
+    public double getDouble(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        return getValue(columnIndex, bsonValue -> bsonValue.asDouble().getValue(), 0)
+                .doubleValue();
+    }
+
+    @Override
+    public byte @Nullable [] getBytes(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException();
```

Comment:
We support `setBytes` in `MongoPreparedStatement`, why do we not support `getBytes` in `MongoResultSet`? I would expect that we either not support this type in both places, or support it in both.

Reply:
I overlooked. I mistook `getBytes()` with `getBinaryStream()`. Thanks for the discovery.